substrate/frame/core-fellowship/src/lib.rs

```diff
@@ -603,3 +608,29 @@ impl_ensure_origin_with_arg_ignoring_arg! {
 		EnsureOriginWithArg<T::RuntimeOrigin, A> for EnsureInducted<T, I, MIN_RANK>
 	{}
 }
+
+impl<T: Config<I>, I: 'static> RankedMembersSwapHandler<T::AccountId, Rank> for Pallet<T, I> {
+	fn swapped(old: &T::AccountId, new: &T::AccountId, _rank: Rank) {
+		if old == new {
+			defensive!("Should not try to swap with self");
+			return
+		}
+		if !Member::<T, I>::contains_key(old) {
+			defensive!("Should not try to swap non-member");
+			return
+		}
+		if Member::<T, I>::contains_key(new) {
+			defensive!("Should not try to overwrite existing member");
+			return
+		}
+
+		if let Some(member) = Member::<T, I>::take(old) {
+			Member::<T, I>::insert(new, member);
+		}
+		if let Some(we) = MemberEvidence::<T, I>::take(old) {
+			MemberEvidence::<T, I>::insert(new, we);
+		}
+
+		Self::deposit_event(Event::<T, I>::Swapped { who: old.clone(), new_who: new.clone() });
```

Comment:
I am itching for a try-state check that ensures members that are kept in `Config::Members` are the same as the set that is kept here. Something like:

```diff
diff --git a/substrate/frame/core-fellowship/src/lib.rs b/substrate/frame/core-fellowship/src/lib.rs
index a0a45c7c59..1db600deb0 100644
--- a/substrate/frame/core-fellowship/src/lib.rs
+++ b/substrate/frame/core-fellowship/src/lib.rs
@@ -547,6 +547,17 @@ pub mod pallet {
 				Self::deposit_event(e);
 			}
 		}
+
+		/// Invariants:
+		///
+		/// - All members tracked in this pallet under [`Members`] and [`MemberEvidence`] must have
+		///   be part of [`Config::Members`].
+		#[cfg(feature = "try-runtime")]
+		fn do_try_state() {
+			let external_members = T::Members::all().collect::<Vec<_>>();
+			let internal_members = todo!();
+			// convert to set, then compare.
+		}
 	}
 
 	impl<T: Config<I>, I: 'static> GetSalary<RankOf<T, I>, T::AccountId, T::Balance> for Pallet<T, I> {
diff --git a/substrate/frame/ranked-collective/src/lib.rs b/substrate/frame/ranked-collective/src/lib.rs
index 88631884c5..92dc44e16f 100644
--- a/substrate/frame/ranked-collective/src/lib.rs
+++ b/substrate/frame/ranked-collective/src/lib.rs
@@ -838,5 +838,10 @@ pub mod pallet {
 		fn demote(who: &Self::AccountId) -> DispatchResult {
 			Self::do_demote_member(who.clone(), None)
 		}
+
+		#[cfg(feature = "try-runtime")]
+		fn all() -> Vec<T::AccountId> {
+			Members::<T, I>::iter().map(|(who, _)| who).collect()
+		}
 	}
 }
diff --git a/substrate/frame/support/src/traits/members.rs b/substrate/frame/support/src/traits/members.rs
index d667eaa7e9..edd627257c 100644
--- a/substrate/frame/support/src/traits/members.rs
+++ b/substrate/frame/support/src/traits/members.rs
@@ -295,6 +295,9 @@ pub trait RankedMembers {
 	/// Demote a member to the next lower rank; demoting beyond the `min_rank` removes the
 	/// member entirely.
 	fn demote(who: &Self::AccountId) -> DispatchResult;
+
+	#[cfg(feature = "try-runtime")]
+	fn all() -> Vec<T::AccountId>;
 }
 
 /// Trait for type that can handle the initialization of account IDs at genesis.
```

Reply:
Will do as follow up